README.md
![jCart Ajax (for Joomshopping)](https://github.com/Kostelano/jCart-Ajax-Joomshopping/blob/master/jcart-ajax-joomshopping.png)

# jCart Ajax (for Joomshopping)
Module for the [Joomshopping](http://webdesigner-profi.de/joomla-webdesign/joomla-shop) component, which allows you to present the cart in a pop-up window, as well as add, reduce and delete products in Ajax mode.

## Project page
[jCart Ajax (for Joomshopping)](http://jnotes.net.ua/ext/jcart-ajax)

## Version
![version](https://img.shields.io/badge/stable-2.0.2-blue?style=for-the-badge) ![version-dev](https://img.shields.io/badge/dev-2.0.3-red?style=for-the-badge)

## System requirements
![joomla-version](https://img.shields.io/badge/joomla-3.9x-green?style=for-the-badge) ![joomshopping-version](https://img.shields.io/badge/joomshopping-4x-green?style=for-the-badge) ![php-version](https://img.shields.io/badge/php-7.3x-green?style=for-the-badge)

## Copyright
Copyright (C) 2015-2018 Sadko, [Sk.ru](http://www.sk.ru). All rights reserved.  
Copyright (C) 2019-2020 Konstantin Kolos, [Jnotes.net.ua](http://www.jnotes.net.ua). All rights reserved.

Distributed under the [GNU General Public License v3](http://www.gnu.org/licenses/gpl-3.0.html)

This project is open source project, feel free to contribute! Thank you.


CHANGELOG.md
# Changelog
All noticeable changes in the module will be documented in this file.

### [2.0.2](https://github.com/Kostelano/jCart-Ajax-Joomshopping/releases/tag/2.0.2) - 01.01.2020
- Added 2 more additional (custom) fields,
- Images, styles and scripts moved to the /media directory,
- Changed the style override logic.

### [2.0.1](https://github.com/Kostelano/jCart-Ajax-Joomshopping/releases/tag/2.0.1) - 29.09.2019
- Module release based on [Multi Basket Ajax](https://joomlaforum.ru/index.php/topic,316647.0.html) (author [Sadko](https://joomlaforum.ru/index.php?action=profile;u=189020)),
- Added parameter: cart icon,
- Added parameter: wishlist icon,
- Added parameter: additional fields (to display other links or pages on the site),
- Added parameter: override CSS stylesheet,
- Added parameter: tooltip (title, data-title),
- Added button to empty shopping cart,
- Added server for automatic search and installation of updates,
- Added languages: english (en-GB), russian (ru-RU), ukrainian (uk-UA).


tmpl/default.php
<?php
/**
 * @package	jCart Ajax
 * @author	Konstantin Kolos
 * @copyright	Copyright (C) 2015-2018 Sadko, http://www.sk.ru
 * @copyright	Copyright (C) 2019-2020 Jnotes.net.ua. All rights reserved
 * @contact	http://jnotes.net.ua, admin@jnotes.net.ua
 * @license	http://gnu.org/licenses/gpl-3.0.html, GNU/GPLv3
**/

defined('_JEXEC') or die('Restricted access');
?>

<style type="text/css">
.jc-img-cart-select {
	background: url(/<?php echo $params->get('cartImageSelect'); ?>) no-repeat scroll 0 0;
	width: <?php echo $params->get('cartImageSelectWidth'); ?>;
	height: <?php echo $params->get('cartImageSelectHeight'); ?>;
}
.jc-img-wish-select {
	background: url(/<?php echo $params->get('wishImageSelect'); ?>) no-repeat scroll 0 0;
	width: <?php echo $params->get('wishImageSelectWidth'); ?>;
	height: <?php echo $params->get('wishImageSelectHeight'); ?>;
}
</style>

<div class="jc-shop">
<?php if ($params->get('cartIcon') == 0) { ?>

	<?php if ($params->get('cartFields_1') == 1) { ?>
		<a class="jc-field" href="<?php echo $params->get('cartFieldUrl_1'); ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo $params->get('cartFieldTitle_1'); ?>" target="<?php echo $params->get('cartFieldsAttribute'); ?>"><?php echo $params->get('cartFieldTitle_1'); ?></a>&nbsp;<?php echo $params->get('cartFieldsSeparator'); ?>
	<?php } ?>

	<?php if ($params->get('cartFields_2') == 1) { ?>
		<a class="jc-field" href="<?php echo $params->get('cartFieldUrl_2'); ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo $params->get('cartFieldTitle_2'); ?>" target="<?php echo $params->get('cartFieldsAttribute'); ?>"><?php echo $params->get('cartFieldTitle_2'); ?></a>&nbsp;<?php echo $params->get('cartFieldsSeparator'); ?>
	<?php } ?>

	<?php if ($params->get('cartFields_3') == 1) { ?>
		<a class="jc-field" href="<?php echo $params->get('cartFieldUrl_3'); ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo $params->get('cartFieldTitle_3'); ?>" target="<?php echo $params->get('cartFieldsAttribute'); ?>"><?php echo $params->get('cartFieldTitle_3'); ?></a>&nbsp;<?php echo $params->get('cartFieldsSeparator'); ?>
	<?php } ?>

	<?php if ($params->get('cartFields_4') == 1) { ?>
		<a class="jc-field" href="<?php echo $params->get('cartFieldUrl_4'); ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo $params->get('cartFieldTitle_4'); ?>" target="<?php echo $params->get('cartFieldsAttribute'); ?>"><?php echo $params->get('cartFieldTitle_4'); ?></a>&nbsp;<?php echo $params->get('cartFieldsSeparator'); ?>
	<?php } ?>

	<?php if ($params->get('cartFields_5') == 1) { ?>
		<a class="jc-field" href="<?php echo $params->get('cartFieldUrl_5'); ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo $params->get('cartFieldTitle_5'); ?>" target="<?php echo $params->get('cartFieldsAttribute'); ?>"><?php echo $params->get('cartFieldTitle_5'); ?></a>&nbsp;<?php echo $params->get('cartFieldsSeparator'); ?>
	<?php } ?>

	<a class="jc-cart-text" href="<?php echo SEFLink('index.php?option=com_jshopping&controller=cart&task=view', 1) ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo JText::_('MOD_JCART_AJAX_GO_TO_CART') ?>"><?php echo JText::_('MOD_JCART_AJAX_CART') ?></a>

	(<a class="jc-qt-product" href="<?php echo SEFLink('index.php?option=com_jshopping&controller=cart&task=view', 1) ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo JText::_('MOD_JCART_AJAX_GO_TO_CART') ?>"><?php echo $cart->count_product ?></a>)

<?php } elseif ($params->get('cartIcon') == 1) { ?>

	<?php if ($params->get('wishIcon') == 1) { ?>
		<a class="jc-img-wish" href="<?php echo SEFLink('index.php?option=com_jshopping&controller=wishlist&task=view', 1) ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo JText::_('MOD_JCART_AJAX_GO_TO_WISHLIST') ?>"><span>&nbsp;</span></a>
	<?php } ?>

	<div class="jc-img-cart">
		<a class="jc-qt-product" href="<?php echo SEFLink('index.php?option=com_jshopping&controller=cart&task=view', 1) ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo JText::_('MOD_JCART_AJAX_GO_TO_CART') ?>"><?php echo $cart->count_product ?></a>
	</div>

<?php } elseif ($params->get('cartIcon') == 2) { ?>

	<?php if ($params->get('wishIcon') == 1) { ?>
		<a class="jc-img-wish-select" href="<?php echo SEFLink('index.php?option=com_jshopping&controller=wishlist&task=view', 1) ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo JText::_('MOD_JCART_AJAX_GO_TO_WISHLIST') ?>"><span>&nbsp;</span></a>
	<?php } ?>

	<div class="jc-img-cart-select">
		<a class="jc-qt-product" href="<?php echo SEFLink('index.php?option=com_jshopping&controller=cart&task=view', 1) ?>" <?php echo $params->get('moduleStyleAttribute'); ?>="<?php echo JText::_('MOD_JCART_AJAX_GO_TO_CART') ?>"><?php echo $cart->count_product ?></a>
	</div>
<?php } ?>

	<div class="jc-content">
	<?php if (count($cart->products) == 0) { ?>
		<div class="jc-list-empty">
			<?php echo JText::_('MOD_JCART_AJAX_CART_IS_EMPTY'), JText::_('MOD_JCART_AJAX_CART_IS_EMPTY_DESC') ?>
		</div>
	<?php } else { ?>

	<div class="jc-list-product">
		<div class="jc-rows">
		<?php foreach ($cart->products as $key_id=>$value) { ?>
			<div class="jc-row">
				<a class="jc-img-product" href="<?php echo $value['href'] ?>">
					<img src="<?php echo $jshopConfig->image_product_live_path ?>/<?php if ($value["thumb_image"]!='') echo $value["thumb_image"]; else echo 'noimage.gif' ?>">
				</a>

				<div class="jc-name-product">
					<a href="<?php echo $value['href'] ?>" name="<?php echo $value["product_id"] ?>"><?php echo $value["product_name"] ?></a>
					<?php foreach ($value['attributes_value'] as $attr) { ?>
						<div class="jc-attr" id="attr_id_<?php echo $value["product_id"].'_'.$attr->attr_id.$attr->value_id ?>">
							<?php echo $attr->attr ?>: <?php echo $attr->value ?>
						</div>
					<?php } ?>
				</div>

				<div class="jc-control">
					<a class="jc-remove" href="<?php echo $value["href_delete"] ?>?ajax=1">&#10006;</a>
						<span class="btn jc-qt-minus" minuskey="quantity[<?php echo $key_id ?>]" minusval="<?php echo $value["quantity"] ?>">&#8722;</span>
							<input class="jc-inputbox" type="text" value="<?php echo $value["quantity"] ?>" name="quantity[<?php echo $key_id ?>]">
						<span class="btn jc-qt-plus" pluskey="quantity[<?php echo $key_id ?>]" plusval="<?php echo $value["quantity"] ?>">&#43;</span>
					<span class="jc-price">&rarr; <?php echo formatprice($value["price"]) ?></span>
				</div>
			</div>
		<?php } ?>
		</div>

		<hr class="jc-hr">
		<div class="jc-total">
			<?php echo JText::_('MOD_JCART_AJAX_TOTAL_PRODUCTS').': <span class="jc-total-qt">'.$cart->count_product.'</span>, '.JText::_('MOD_JCART_AJAX_TOTAL_AMOUNT').' <span class="jc-total-qt">'.formatprice($cart->getSum(0, 1)) ?></span>
		</div>

		<div class="jc-btn-group">
			<a class="btn btn-primary" href="<?php echo SEFLink('index.php?option=com_jshopping&controller=cart&task=view', 1) ?>"><?php echo JText::_('MOD_JCART_AJAX_GO_TO_CART') ?></a>
			<a class="btn btn-success" href="<?php echo SEFLink('index.php?option=com_jshopping&controller=cart&task=clear', 1); ?>" onclick="return confirm('<?php echo JText::_('MOD_JCART_AJAX_CLEAR_CART_DESC') ?>')"><?php echo JText::_('MOD_JCART_AJAX_CLEAR_CART') ?></a>
		</div>
	</div>

	<?php } ?>
	</div>
</div>

<div class="jc-hidden">
	<span class="datac-one"><?php echo JURI::base() ?></span>
	<span class="datac-uic"><?php echo SEFLink('index.php?option=com_jshopping&controller=cart&task=view', 1) ?></span>
	<span class="datac-urp"><?php echo SEFLink('index.php?option=com_jshopping&controller=product&task=view', 1) ?></span>
	<span class="datac-udl"><?php echo SEFLink('index.php?option=com_jshopping&controller=cart&task=delete', 1) ?></span>
	<span class="datac-ugx"><?php echo SEFLink('index.php?option=com_jshopping&controller=cart&task=clear', 1) ?></span>
	<span class="datac-tip"><?php echo $jshopConfig->image_product_live_path ?></span>
	<span class="datac-vpp"><?php echo $jshopConfig->product_price_precision ?></span>
	<span class="datac-hcc"><?php echo $jshopConfig->currency_code ?></span>
	<span class="datac-tac"><?php echo JText::_('MOD_JCART_AJAX_PRODUCT_IN_CART') ?></span>
	<span class="datac-tgc"><?php echo JText::_('MOD_JCART_AJAX_GO_TO_CART') ?></span>
	<span class="datac-tpc"><?php echo JText::_('MOD_JCART_AJAX_TOTAL_PRODUCTS') ?></span>
	<span class="datac-tps"><?php echo JText::_('MOD_JCART_AJAX_TOTAL_AMOUNT') ?></span>
	<span class="datac-tgx"><?php echo JText::_('MOD_JCART_AJAX_CLEAR_CART') ?></span>
	<span class="datac-err"><?php echo JText::_('MOD_JCART_AJAX_ERROR_ADDING_TO_CART') ?></span>
</div>


mod_jcart_ajax.php
<?php
/**
 * @package	jCart Ajax
 * @author	Konstantin Kolos
 * @copyright	Copyright (C) 2015-2018 Sadko, http://www.sk.ru
 * @copyright	Copyright (C) 2019-2020 Jnotes.net.ua. All rights reserved
 * @contact	http://jnotes.net.ua, admin@jnotes.net.ua
 * @license	http://gnu.org/licenses/gpl-3.0.html, GNU/GPLv3
**/

defined('_JEXEC') or die('Restricted access');

error_reporting(error_reporting() & ~E_NOTICE);
if (!file_exists(JPATH_SITE . '/components/com_jshopping/jshopping.php')) {
	JError::raiseError(500, JText::_('MOD_JCART_AJAX_ERROR_COMPONENT_NOT_INSTALLED'));
}
	require_once (JPATH_SITE . '/components/com_jshopping/lib/factory.php');
	require_once (JPATH_SITE . '/components/com_jshopping/lib/functions.php');

	JSFactory::loadCssFiles();
	JSFactory::loadLanguageFile();

	$doc = JFactory::getDocument();

	$doc->addStyleSheet(JURI::base() . 'media/mod_jcart_ajax/css/' . $params->get('moduleStyle'));
	$doc->addScript(JURI::base() . 'media/mod_jcart_ajax/js/jcart-ajax.js');

	$jshopConfig = JSFactory::getConfig();
	JModelLegacy::addIncludePath(JPATH_SITE . '/components/com_jshopping/models');

	$cart = JModelLegacy::getInstance('cart', 'jshop');
	$cart->load("cart");
	$cart->addLinkToProducts(1, $type="cart");

	require JModuleHelper::getLayoutPath('mod_jcart_ajax');
?>


media/js/jcart-ajax.js
jQuery(document).ready(function(){
one=jQuery('span.datac-one').html();
uic=jQuery('span.datac-uic').html();
urp=jQuery('span.datac-urp').html();
udl=jQuery('span.datac-udl').html();
ugx=jQuery('span.datac-ugx').html();
tip=jQuery('span.datac-tip').html();
vpp=jQuery('span.datac-vpp').html();
hcc=jQuery('span.datac-hcc').html();
tac=jQuery('span.datac-tac').text();
tgc=jQuery('span.datac-tgc').text();
tpc=jQuery('span.datac-tpc').text();
tps=jQuery('span.datac-tps').text();
tgx=jQuery('span.datac-tgx').text();
err=jQuery('span.datac-err').text();
if(jQuery('form[name="updateCart"], form[name="oneStepCheckoutForm"]').length){
	jQuery('.jc-shop .jc-content').remove();
}
function Af(){
	jQuery('.jc-in-cart-list').remove();
	jQuery('.jc-in-cart').remove();
	a=jQuery('input#product_id').val();
	b=jQuery('.jc-shop .jc-name-product a').map(function(){return jQuery(this).attr('name')});
	jQuery.each(jQuery.unique(b),function(b,c){
	jQuery('.productitem_'+c+' .product_buy, .productitem_'+c+' .btn-success').after('<a class="jc-in-cart-list" href="'+uic+'">'+tac+'</a>');
	a==c&&jQuery('.prod_buttons .buttons').after('<a class="btn btn-success jc-in-cart" href="'+uic+'">'+tac+'</a>');
	});
}
function Bf(a){Af(),
	jQuery('.productitem_'+a+' .jc-in-cart-list').prepend('<div class="jc-list-added">'+tac+'</div>'),
	jQuery('.jc-in-cart').prepend('<div class="jc-in-cart-added">'+tac+'</div>'),
	jQuery('.jc-list-added, .jc-in-cart-added').fadeOut(2400)
}
function Cf(a){for(var b in a)if('products'==b&&a.hasOwnProperty(b))for(var c in a[b])if(a[b].hasOwnProperty(c)){
	for(var d in a[b][c]){
		'category_id'==d&&(str_cat=a[b][c][d]);
		'product_id'==d&&(str_pid=a[b][c][d]);
		'thumb_image'==d&&(str_img=a[b][c][d]);
		if(str_img=='')var str_img='noimage.gif';
		'product_name'==d&&(str_pro=a[b][c][d]);
		if('attributes_value'==d){var str_atr='';for(e in a[b][c][d]) if(a[b][c][d].hasOwnProperty(e)){
				for(var f in a[b][c][d][e])
				'attr_id'==f&&(str_aid=a[b][c][d][e][f]),
				'value_id'==f&&(str_vid=a[b][c][d][e][f]),
				'attr'==f&&(str_anm=a[b][c][d][e][f]),
				'value'==f&&(str_avl=a[b][c][d][e][f]);
				str_atr+='<div class="jc-attr" id="attr_id_'+str_pid+'_'+str_aid+str_vid+'">'+str_anm+': '+str_avl+'</div>'
		}}
		if('quantity'==d){
			str_but='<span class="btn jc-qt-minus" minuskey="quantity['+c+']" minusval="'+a[b][c][d]+'">&#8722;</span> <input type="text" value="'
			+a[b][c][d]+'" class="jc-inputbox" name="quantity['+c+']"> <span class="btn jc-qt-plus" pluskey="quantity['+c+']" plusval="'+a[b][c][d]+'">&#43;</span>';
		}
		'price'==d&&(str_prc='<span class="jc-price">&rarr; '+a[b][c][d].toFixed(vpp)+' '+hcc+'</span>')
	}
	str_row+='<div class="jc-row"><a class="jc-img-product" href="'+urp+'?category_id='+str_cat+'&product_id='+str_pid+'"><img src="'
	+tip+'/'+str_img+'"></a><div class="jc-name-product"><a href="'+urp+'?category_id='+str_cat+'&product_id='+str_pid+'" name="'+str_pid+'">'
	+str_pro+'</a>'+str_atr+'</div><div class="jc-control"><a class="jc-remove" href="'+udl+'?number_id='+c+'?ajax=1">&#10006;</a> '+str_but+''+str_prc+'</div></div>'
}}
function Df(a){
	str_row='';Cf(a);if((a.price_product-a.rabatt_summ)>0)str_sum=(a.price_product-a.rabatt_summ).toFixed(vpp);else str_sum=0;
	jQuery('.jc-shop .jc-content').html('<div class="jc-list-product"><div class="jc-rows">'+str_row+'</div><hr class="jc-hr"><div class="jc-total">'+tpc+': <span class="jc-total-qt">'+a.count_product+'</span>, '+tps+' <span class="jc-total-qt">'+str_sum+' '+hcc+'</span></div><div class="jc-btn-group"><a class="btn btn-primary" href="'+uic+'">'+tgc+'</a> <a class="btn btn-success" href="'+ugx+'">'+tgx+'</a></div></div>')
}
jQuery('body').on('click','.oiproduct .button_buy, .product_corps .button_buy',function(a){
	a.preventDefault();
	b=jQuery(this).attr('href');
	c=b.split("product_id=")[1];
	// jQuery('body').append('<div class="jc-loading-ajax"></div>');
	jQuery.ajax({cache:!1,url:b+'&ajax=1',dataType:'json',success:function(a){
		jQuery('.jc-loading-ajax').remove();
		'cart'==a.type_cart?(jQuery('.jc-shop .jc-qt-product').html(a.count_product),Df(a),Bf(c)):window.location.assign(b)
	},error:function(a){jQuery('.jc-loading-ajax').remove();location.reload()}});return!1
});
jQuery('body').on('click','.prod_buttons .prod_buy, .prod_buttons .btn-primary',function(a){
	a.preventDefault();
	a=jQuery('form[name="product"]').serialize();
	// jQuery('body').append('<div class="jc-loading-ajax"></div>');
	jQuery.ajax({cache:!1,url:one+'index.php?option=com_jshopping&controller=cart&task=add&'+a+'&ajax=1',dataType:'json',ifModified:true,success:function(a){
		jQuery('.jc-loading-ajax').remove();
		'cart'==a.type_cart?(jQuery('.jc-shop .jc-qt-product').html(a.count_product),Df(a),Bf()):alert(err)
	},error:function(){jQuery('.jc-loading-ajax').remove();location.reload()}});return!1
});
jQuery('body').on('click','.jc-remove',function(a){
	a.preventDefault();
	a=jQuery(this).attr('href');
	jQuery('.jc-shop .jc-total').append('<div class="jc-loading"></div>');
	jQuery.ajax({cache:!1,url:a+'&ajax=1',dataType:'json',success:function(a){
		jQuery('.jc-loading').remove();
		'cart'==a.type_cart?(
			jQuery('.jc-shop .jc-qt-product').html(a.count_product),str_row='',Cf(a),
			str_sum=(a.price_product-a.rabatt_summ).toFixed(vpp),str_sum<0&&(jQuery(str_sum=0)),
			jQuery('.jc-shop .jc-content').html('<div class="jc-list-product"><div class="jc-rows">'+str_row+'</div><hr class="jc-hr"><div class="jc-total">'
			+tpc+': '+a.count_product+', '+tps+': '+str_sum+' '+hcc+'</div><div class="jc-btn-group"><a class="btn btn-primary" href="'+uic+'">'+tgc+'</a> <a class="btn btn-success" href="'+ugx+'">'+tgx+'</a></div></div>'),
			a.count_product=='0'&&(jQuery('.jc-shop .jc-list-product').remove()),Af()
		):setTimeout(function(){location.reload()})
	},error:function(){jQuery('.jc-loading').remove();location.reload()}});return!1
});
jQuery('body').on('click','.jc-qt-minus',function(a){
	a.preventDefault();
	a=jQuery(this).attr('minuskey');
	b=jQuery(this).attr('minusval'),c=parseFloat(b)-1;
	a=0!=c?a+'='+c:a+'='+b;
	jQuery('.jc-shop .jc-total').append('<div class="jc-loading"></div>');
	jQuery.ajax({cache:!1,url:one+'index.php?option=com_jshopping&controller=cart&task=refresh&'+a+'&ajax=1',dataType:'json',ifModified:true,success:function(a){
		jQuery('.jc-loading').remove();
		'cart'==a.type_cart&&(jQuery('.jc-shop .jc-qt-product').html(a.count_product),Df(a))
	},error:function(){jQuery('.jc-loading').remove();location.reload()}});return!1
});
jQuery('body').on('click','.jc-qt-plus',function(a){
	a.preventDefault();
	a=jQuery(this).attr('pluskey');
	b=jQuery(this).attr('plusval'),b=parseFloat(b)+1;
	a=a+'='+b;
	jQuery('.jc-shop .jc-total').append('<div class="jc-loading"></div>');
	jQuery.ajax({cache:!1,url:one+'index.php?option=com_jshopping&controller=cart&task=refresh&'+a+'&ajax=1',dataType:'json',ifModified:true,success:function(a){
		jQuery('.jc-loading').remove();
		'cart'==a.type_cart&&(jQuery('.jc-shop .jc-qt-product').html(a.count_product),Df(a))
	},error:function(){jQuery('.jc-loading').remove();location.reload()}});return!1
});
jQuery('body').on('keyup','.jc-inputbox',function(a){
	a.preventDefault();
	a=jQuery(this).attr('name');
	b=jQuery(this).val();
	if(0!=b)a=a+'='+b,
	jQuery('.jc-shop .jc-total').append('<div class="jc-loading"></div>'),
	jQuery.ajax({cache:!1,url:one+'index.php?option=com_jshopping&controller=cart&task=refresh&'+a+'&ajax=1',dataType:'json',ifModified:true,success:function(a){
		jQuery('.jc-loading').remove();
		'cart'==a.type_cart?(jQuery('.jc-shop .jc-qt-product').html(a.count_product),Df(a)):setTimeout(function(){location.reload()})
	},error:function(){jQuery('.jc-loading').remove();location.reload()}});else return!1
});Af()});
